CodingTracker.TildaDares/Infrastructure/CodingGoalsDatabase.cs

```diff
@@ -0,0 +1,204 @@
+using System.Configuration;
+using CodingTracker.TildaDares.Models;
+using Dapper;
+using Microsoft.Data.Sqlite;
+using Spectre.Console;
+
+namespace CodingTracker.TildaDares.Infrastructure;
+
+public class CodingGoalsDatabase
+{
+    private readonly string _connectionString = ConfigurationManager.ConnectionStrings["CodingGoalDB"].ConnectionString;
+    
+    public CodingGoalsDatabase()
+    {
+        CreateCodingGoalDb();
+    }
+
+    public void InsertCodingGoal(CodingGoal codingGoal)
+    {
+        using var connection = new SqliteConnection(_connectionString);
+        try
+        {
+            connection.Open();
+            const string sql = "INSERT INTO codingGoal(startTime, endTime, totalHoursGoal) VALUES (@StartTime, @EndTime, @TotalHoursGoal)";
+            var rowsAffected = connection.Execute(sql, codingGoal);
+            AnsiConsole.MarkupLine($"[green]{rowsAffected} row(s) inserted.[/]");
```

Comment:
🟠 I would advise that here you have intertwined your Infrastructure layer with your Presentation layer here.

🔧 Maybe a refactor to return a message or bool? and handle the message display after the call to this function.

Reply:
Excellent point!